AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/Keyrings/AwsKms/AwsKmsHierarchicalKeyring.dfy

```diff
@@ -161,6 +161,10 @@ module AwsKmsHierarchicalKeyring {
       //= type=implication
       //# - MAY provide a max cache size
       maxCacheSize: Types.PositiveInteger,
+      gracePeriod: Types.PositiveInteger,
+      graceInterval: Types.PositiveInteger,
+      fanOut: Types.PositiveInteger,
+      inFlightTTL: Types.PositiveInteger,
```

Comment:
Having these be properties of the operation make sense.
But would it be cleaner to have the class actually take the constructed CMC?

Reply:
I don't know. I don't feel qualified to make such a decision.

Reply:
I think that making the trackerSettings a structure maybe was enough.